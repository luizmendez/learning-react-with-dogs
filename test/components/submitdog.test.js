import React from 'react';
import SubmitDog from 'components/submitdog/submitdog';

const formElements = [
    {
        name: 'dogName',
        value: 'Terry',
        type: 'input',
        getAttribute: () => 'input'
    },
    {
        name: 'dogBreed',
        value: 'airedale',
        type: 'select',
        getAttribute: () => 'select'
    },
    {
        name: 'dogPic',
        type: 'file',
        getAttribute: () => 'file',
        files: [
            {
                name: test.jpg
            }
        ]
    }
];

describe('<SubmitDog />', () => {
    it('renders the SubmitDog component', () => {
        const wrapper = shallow(<SubmitDog />);
        expect(wrapper.find('.submit-dog-content').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('submits the form', () => {
        const sendDogForm = jest.fn();
        const wrapper = shallow(<SubmitDog sendDogForm={sendDogForm} />);
        const form = wrapper.find('.form-dog').first();
        const preventDefault = jest.fn();
        const e = {
            preventDefault,
            target: { elements: formElements }
        };
        form.simulate('submit', e);
        expect(e.preventDefault).toBeCalledTimes(1);
        expect(sendDogForm).toBeCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });
});
