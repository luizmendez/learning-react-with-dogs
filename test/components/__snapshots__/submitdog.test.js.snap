// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitDog /> renders the SubmitDog component 1`] = `
<div
  className="submit-dog-content"
>
  <h2>
    Submit a Dog Pic
  </h2>
  <p>
    To submit a dog pic, fill all the fields below and submit the form.
  </p>
  <div
    className="row"
  >
    <form
      className="col-md-6 form-dog"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="submitDogName"
        >
          Dog's Name
        </label>
        <input
          className="form-control"
          id="submitDogName"
          name="dogName"
          placeholder="Ramona"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="submitDogBreed"
        >
          Dog's Breed
        </label>
        <select
          className="form-control"
          id="submitDogBreed"
          name="dogBreed"
          required={true}
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Dog's Picture
        </label>
        <div
          className="custom-file"
        >
          <input
            accept="image/*"
            className="custom-file-input"
            id="customFile"
            name="dogPic"
            onChange={[Function]}
            required={true}
            type="file"
          />
          <label
            className="custom-file-label"
            htmlFor="customFile"
          >
            Choose File
          </label>
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      className="pic-preview col-md-6"
    >
      <img
        alt=""
      />
    </div>
  </div>
</div>
`;

exports[`<SubmitDog /> submits the form 1`] = `
<div
  className="submit-dog-content"
>
  <h2>
    Submit a Dog Pic
  </h2>
  <p>
    To submit a dog pic, fill all the fields below and submit the form.
  </p>
  <div
    className="row"
  >
    <form
      className="col-md-6 form-dog"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="submitDogName"
        >
          Dog's Name
        </label>
        <input
          className="form-control"
          id="submitDogName"
          name="dogName"
          placeholder="Ramona"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="submitDogBreed"
        >
          Dog's Breed
        </label>
        <select
          className="form-control"
          id="submitDogBreed"
          name="dogBreed"
          required={true}
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Dog's Picture
        </label>
        <div
          className="custom-file"
        >
          <input
            accept="image/*"
            className="custom-file-input"
            id="customFile"
            name="dogPic"
            onChange={[Function]}
            required={true}
            type="file"
          />
          <label
            className="custom-file-label"
            htmlFor="customFile"
          >
            Choose File
          </label>
        </div>
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      className="pic-preview col-md-6"
    >
      <img
        alt=""
      />
    </div>
  </div>
</div>
`;
